### PR TITLE
feat: filter expired tiles by geometry

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: src/cli/commands/append/appendManager.ts [hunk 1]
```diff
@@ -230,4 +256,35 @@ export class AppendManager {
...
+      const isValid = geojsonValidator(geojson);
+      if (!isValid) {
+        throw new InvalidGeojsonError(`geojson with key ${resource.key} is invalid`);
+      }
+
+      geometries[resource.key] = geojson;
+    });
+
+    await Promise.all(getResourcePromises);
+  }
```
Comment (syncush) [inline src/cli/commands/append/appendManager.ts:289]: I think the logic of remote resources from S3 should not be included inside appendManager, also the assertion of the geojson should be in a separate module.

Moreover, the app should retrieve the geojson file on startup, so maybe it should be injected into the app while bootstrapping, @CptSchnitz wdyt?

File: src/cli/commands/append/expireTilesFilters.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            [east, north],
+            [west, north],
+            [west, south],
+          ],
+        ],
+      };
+      return booleanContains(geometry, bboxAsPolygon);
+    });
+  };
+  return filter;
```
Comment (syncush) [inline src/cli/commands/append/expireTilesFilters.ts:37]: if it efficient since you are transforming the bbox to polygon?

**Review by syncush — CHANGES_REQUESTED** [review]

File: src/cli/commands/append/appendManager.ts (hunk 1 above)
Comment (syncush) [reply src/cli/commands/append/appendManager.ts:289]: why would we share the same configuration for multiple services? in our use case a .geojson file should be fetched at startup describing the area for relevant tiles

File: src/cli/commands/append/expireTilesFilters.ts (hunk 2 above)
Comment (melancholiai) [reply src/cli/commands/append/expireTilesFilters.ts]: I've improved the filtering chain functionality,
we get the w\s\e\n with `@mapbox/sphericalmercator` `bbox` function.
it is possible filter the tile before creating the whole bbox array but it losses the generic functionality of building the bbox and filtering them separately, for instance, say in the future we'd like to match some filters and not all.
also, it just seems less readable

File: src/cli/commands/append/appendManager.ts (hunk 1 above)
Comment (melancholiai) [reply src/cli/commands/append/appendManager.ts]: the fetching is now done in another module.

**Review by syncush — APPROVED** [review]

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

**Review by syncush — APPROVED** [review]

**Pull request merged**